ecosystem/typescript/sdk_v2/src/bcs/deserializer.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable no-bitwise */
+import { MAX_U32_NUMBER } from "./consts";
+import { Uint128, Uint16, Uint256, Uint32, Uint64, Uint8 } from "./types";
+
+// The class must implement a static deserialize method.
+interface Deserializable<T> {
+  deserialize(deserializer: Deserializer): T;
+}
+
+export class Deserializer {
+  private buffer: ArrayBuffer;
+
+  private offset: number;
+
+  constructor(data: Uint8Array) {
```

Comment:
Curious - do we want to use `Uint8Array` here or `HexInput`? I am guessing we will stick with `Uint8Array` for any data/buffer related?

Reply:
yeah I'll stick to `Uint8Array` for now, just to keep this all the same. I think we can address whether or not it should change in next meeting. And then address the lack of object args, too